src/backends/compose/compose_backend.py

```diff
@@ -177,18 +177,18 @@ def get_file(self, tank_index: int, service: ServiceType, file_path: str):
         out = out[: stat["size"]]
         return out
 
-    def get_tank_ipv4(self, index: int) -> str:
+    def get_tank_ipv4(self, index: int) -> str | None:
         c = self.get_container(index, ServiceType.BITCOIN)
         if c:
             return self.get_ipv4_address(c)
         else:
             return None
 
-    def get_tank_ip_addr(self, index: int) -> str:
-        self.get_tank_ipv4(index)
+    def get_tank_ip_addr(self, index: int) -> str | None:
+        return self.get_tank_ipv4(index)
 
-    def get_tank_dns_addr(self, index: int) -> str:
-        self.get_tank_ip_addr(index)
+    def get_tank_dns_addr(self, index: int) -> str | None:
+        return self.get_tank_ip_addr(index)
```

Comment:
crap how'd we miss this? how did this pass the tests?

Reply:
The tests never call that function!

The other thing I noticed was that the dag_connect test does not call `debug_assert_log`. I think that means that as long as the test runs, it passes. Need to stitch the tests up.